### PR TITLE
feat(auth-server): Convert subscriptionPaymentExpired + subscriptionsPaymentExpired to mjml stack

### LINKED ISSUE #9278: [emails M4] Convert `subscriptionPaymentExpired`
Take `subscriptionPaymentExpired.html` and any needed partials and convert the template(s) over to the new emails stack. Also convert tests, add states to the state previewing solution, and ensure the template is localized. Make sure it builds properly and that the new email is sent locally by turning on the feature flag it sits behind.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-3548)

### LINKED ISSUE #9296: [emails M4] Convert `subscriptionsPaymentExpired`
Take `subscriptionsPaymentExpired.html` and any needed partials and convert the template(s) over to the new emails stack. Also convert tests, add states to the state previewing solution, and ensure the template is localized. Make sure it builds properly and that the new email is sent locally by turning on the feature flag it sits behind.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXA-3554)

### PR DESCRIPTION
**Pull request by LZoog**
~This is 95% done but I'm running into a Typescript-related issue with new tests for these templates and need to circle back early next week. That's all that's remaining though.~

fixes #9296
fixes #9278

To check this out:
* run `yarn storybook-mjml` in `fxa-auth-server`
* also run `yarn write-emails` to write the old templates to disk. You'll only see `subscriptionPaymentExpired` and can compare the HTML and plaintext version to the one you see in Storybook (minus URLs with query params - it's my hope later we can sort of retire `write-emails` if we can generate the actual query params for display in SB). If you update `write-emails-to-disk` `message.subscriptions` to hold multiple objects, you can run `yarn write-emails` again to see `subscriptionsPaymentExpired`.
* you can also update `mjml.enabledEmailAddress` in the dev config to `@testuser.com` and run the `yarn write-emails` command to see new templates written to disk

### CODE REVIEW

File: packages/fxa-auth-server/test/local/senders/mjml-emails.ts [hunk 1]
```diff
@@ -968,6 +980,45 @@ const TESTS: [string, any][] = [
...
+  ['subscriptionPaymentExpiredEmail', new Map<string, Test | any>([
+    ['subject', { test: 'equal', expected: `Credit card for ${MESSAGE.productName} expiring soon` }],
+    ['headers', new Map([
+      ['X-SES-MESSAGE-TAGS', { test: 'equal', expected: sesMessageTagsHeaderValue('subscriptionPaymentExpired') }],
+      ['X-Template-Name', { test: 'equal', expected: 'subscriptionPaymentExpired' }],
+      ['X-Template-Version', { test: 'equal', expected: TEMPLATE_VERSIONS.subscriptionPaymentExpired }],
+    ])],
+    ['html', [
+      { test: 'include', expected: decodeUrl(configHref('subscriptionSettingsUrl', 'subscription-payment-expired', 'update-billing', 'plan_id', 'product_id', 'uid', 'email')) },
+      // { test: 'include', expected: decodeUrl(configHref('subscriptionTermsUrl', 'subscription-payment-expired', 'subscription-terms')) },
```
Comment (jodyheavener) [inline packages/fxa-auth-server/test/local/senders/mjml-emails.ts:993]: So.. I couldn't find this link in the current/old templates, let alone the new one. Any idea what this is testing?

Comment (LZoog) [reply packages/fxa-auth-server/test/local/senders/mjml-emails.ts:993]: Hmm, looks like in the config it's noted as the cancellation policy URL, do the old templates no longer reference that (I wonder why the old test passes then)?

**Comment by jodyheavener** [comment]
@LZoog I think I got everything, save for the one test comment I wasn't sure about. If you wanna give it your own sign off I can also R+ and merge

**Comment by LZoog** [comment]
@jodyheavener I rebased the branch against `main`, updated the `en-US.ftl` files to `en.ftl`, and added one missing template to `dev.json`. I started looking into that test you mentioned, then noticed that a different one was failing from adding that template to `dev.json` so I just fixed that with another push.

About that test you commented out, looks like it's looking for `https://www.mozilla.org/about/legal/terms/firefox-private-network/<utm params>` and I do see that one in our templates as `https://www.mozilla.org/about/legal/terms/services/` in the subplat layout file..

```
<a class="link-blue" data-l10n-id="fxa-service-url" href="https://www.mozilla.org/about/legal/terms/services/">Firefox Cloud Terms of Service</a>
```

I think this might need to come from the server similar to how we're rendering `privacyUrl`.

I'm not sure why the new templates aren't rendering for me with `yarn write-emails` after updating the email to `testuser+testbox@testuser.com` in `mjml.enabledEmailAddress`. Will you try again on this branch in the morning? 🤔 I'd like to update the docs and it seems like I'm missing something.

**Review by jodyheavener — APPROVED** [review]
sweet, cool, awesome

**Pull request merged**